### PR TITLE
Detailed product page

### PR DESCRIPTION
**Pull request by evfinkn**
This pull request adds

- a detailed product page (see below),
- categories to the new product form,
- pictures to products, and
- adding products to cart.

The detailed product page displays the image of the product, its categories, a link to the seller's profile, the seller's rating (displayed as stars), the number of reviews the seller has, the condition, quantity, price, description, and an "Add to Cart" button, all styled using Bootstrap. The quantity is only displayed if it's greater than 1, and it's displayed with an input to select how many to add to your cart when that button is clicked.

Here's an example product page (the description of which is taken from Wikipedia):

Screenshot 2023-11-24 at 21-17-11 Selt2023FinalProjectTeam001

### CODE REVIEW

**Comment by blhohmann** [comment]
General comments for a future iteration:
I think we could make it so when you click the start on a products page it will auto give that rating to this product/seller or w/e

You can input higher than the available quantity and it will silently downgrade the quantity to the max, maybe sometime we could add a flash saying something like hey the full amount wasn't available so we gave you the max
image
image

File: app/controllers/carts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class CartsController < ApplicationController
+  before_action :require_login
+
+  def add_to_cart
+    product = Product.find params[:product_id]
+    cart = Cart.find_or_initialize_by(
+      product_id: product.id, user: Current.user
+    )
+    cart.quantity ||= 0
```
Comment (blhohmann) [inline app/controllers/carts_controller.rb:9]: I added a puts(cart.quantity) and it is 1 even for products not yet in the cart causing any addition of a product to cart to be one higher than it should be... I'm not sure the root cause of this bug sorry.

See https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/pull/34#issuecomment-1826206467

**Review by blhohmann — CHANGES_REQUESTED** [review]
Looks really really pretty. Great work! Ran into a few small bugs, and noting in a comment things that could be added in the future when we have more time.

**Comment by powersaudrey25** [comment]
> General comments for a future iteration: I think we could make it so when you click the start on a products page it will auto give that rating to this product/seller or w/e
> 
> You can input higher than the available quantity and it will silently downgrade the quantity to the max, maybe sometime we could add a flash saying something like hey the full amount wasn't available so we gave you the max image image
> 
> Not sure what caused this but I tried adding -1, it said I couldn't (which is good) but then when I put in 80, I had 81 in cart? Ignore this unless I can replicate it again or you can. It's possible something else caused it like I already had 1 in the cart or something. image image Update I replicated it again, it always adds one more than actually was input.
> 
> In the future maybe we should add that you have to ctrl+click to get multiple for our non-tech savvy users. image
> 
> Maybe more rspec tests

This might be because the default value for quantity is 1, it is probably adding the chosen quantity plus that default

**Review by blhohmann — APPROVED** [review]
Looks great!

**Pull request merged**